### PR TITLE
Added Quick picker and list command +more

### PR DESCRIPTION
**Pull request by javalight**
These are the changes in this pull request:
- Quick pick/searching edit history
- Persistent storage of edits
- Remove/prune duplicate edits
- Added command to remove,create and toggle edits at cursor location

This create a new kind of workflow where its possible to create an edit at a location to bookmark it, the created edit will move to top of stack. Bookmarking with stack navigation. With picker looking at edits in a list makes going back much easier.

Quick pick demo
Quickpick demo

Search Edits
Quickpick demo search

### CODE REVIEW

**Comment by pgsandstrom** [comment]
Looks good. I'll try to take a closer look and test it within a few days.

**Comment by javalight** [comment]
@pgsandstrom sorry got carried away and just couldn't stop :).

**Comment by pgsandstrom** [comment]
Much appreciated, but it would have been easier to overview if it was split up into more PRs. Some questions:

1. Did you make the logo yourself? It looks good, I just want to be sure I dont intrude on any licenses.

2. Im unsure about the "moveCursorCancelNavigation" command. If I navigate a few steps back in history it can be used to return to the latest edit. But if I havent navigated back a few steps, it does nothing. Shouldn't it always return to the latest edit, or what use case is it meant to fulfill?

**Comment by javalight** [comment]
1. https://icons8.com/icons/set/play, looks like it might be a problem, lets look for one public domain
2. Used to simulate canceling a jump sequence, same as when using the symbols command pallet, on cancel will return to first location before jump, if there is no jump sequence then do nothing.  its mostly used when "top stack on move" is enabled, this way its possible to go back to start with out triggering the current position to move to top of stack. If the user wants to move to top of stack position then goToNextEdit gets the job done.  

Right now i have it bound to escape, in a macro that runs multiple commands on escape. Im using Vim plugin with vscode, making this a very convenient command. Its possible there is a use case for a "moveCursorFirst", maybe lets add both? Matching anyones needs :). 

In short its an cancel command not a move command, maybe should be renamed. 

Todo: add when clause "navigateEditHistory.active" for keybindings, can bind keys when navigation is active

Sorry about the single PR, got addicted, didnt want to stop. lol 

Overall the old behavior has not changed, all changes in behavior need to be enabled in the config.   Besides pruning.

**Comment by javalight** [comment]
licensing: https://icons8.com/license

**Comment by pgsandstrom** [comment]
1. You could also just add a oneliner reference at the bottom of the readme to satisfy the license, I would be okay with either, or simply removing the logo and taking it in another PR.

2. That's a cool use case. Perhaps it could just be clarified in its description that it only has an effect if the user currently has jumped backwards in history.

File: src/extension.ts [hunk 1]
```diff
@@ -105,10 +105,7 @@ export function activate(context: vscode.ExtensionContext) {
     }
 
     const line = range.start.line
-    const lineText =
-      vscode.window.activeTextEditor !== undefined
-        ? vscode.window.activeTextEditor.document.lineAt(line).text.trim()
-        : 'Text Not Found'
+    const lineText = text.trim()
```
Comment (pgsandstrom) [inline src/extension.ts]: Now the editlist contains the changes instead of the entire line. So if I write "hello" then the edit list will just container "o" instead of the entire line. Was that what you intended? It looks kind of weird, imo.

**Comment by pgsandstrom** [comment]
Im pretty sure the check on line 113 can be skipped now, right? The one that checks for multiple edits on the same line.

File: src/extension.ts [hunk 2]
```diff
@@ -202,63 +244,213 @@ export function activate(context: vscode.ExtensionContext) {
...
+
+    const editor = vscode.window.activeTextEditor
+    const currentLine: number = editor ? editor.selection.active.line : 0
+    const currentCharacter: number = editor ? editor.selection.active.character : 0
+    const currentFilePath: string = editor ? editor.document.uri.path : ''
+
+    // push the items
+    const items: QuickPickEdit[] = editList
+      .map((edit: Edit) => {
+        const discription = '(' + edit.line + ')'
```
Comment (pgsandstrom) [inline src/extension.ts]: This line seems to have a of-by-one-error, since lines are 1-indexed in vscode

File: src/extension.ts (hunk 1 above)
Comment (javalight) [reply src/extension.ts]: ah... fixed this localy and never pushed!!

File: src/extension.ts [hunk 3]
```diff
@@ -30,43 +31,61 @@ export function activate(context: vscode.ExtensionContext) {
...
-
       if (e.contentChanges.length === 0) {
         return
       }
 
-      // we only use the last content change, because often that seems to be the relevant one:
-      const lastContentChange = e.contentChanges[e.contentChanges.length - 1]
-      handleContentChange(lastContentChange, filepath)
+      // iterate over all changes, nessisary to keep old edits line numbers up to date
+      e.contentChanges.forEach((change) => addEdit(change.text, change.range, e.document))
```
Comment (pgsandstrom) [inline src/extension.ts:72]: I've previously had some problems with this when triggering stuff like auto formatting that could result in a bunch of edits. But now that I tested it wasn't that bad. So lets try a release with this, and see how it works out.

**Comment by pgsandstrom** [comment]
I'll merge this now. I want to do some housekeeping before I release a new version, but hopefully that will be done within a few days.

Thank you for the contribution!
